crates/sui-json-rpc/src/axum_router.rs

```diff
@@ -144,9 +147,14 @@ async fn process_request<L: Logger>(
     let conn_id = 0; // unused
 
     let params = Params::new(req.params.map(|params| params.get()));
-    let name = rpc_router.route(&req.method, api_version);
+    let name = rpc_router.route(&req.method, api_version, client_addr);
+    // TODO: here we need to somehow insert the client address into the params
+    // (or elsewhere?) depending on the method name
+
```

Comment:
Was a bit unsure how to do this @bmwill .. seems I need to insert the extra arg (client IP) into the `params`, which seems to be some serialized type, potentially a json string. Is that correct? Do you know of a better way?

Reply:
yeah iirc params is just some json at this point so we'd likely need to deserialize, insert, then re-serialize... I told you it would feel icky :D